burr/tracking/client.py

```diff
@@ -85,10 +85,7 @@ def __init__(
 
     @classmethod
     def get_storage_path(cls, project, storage_dir) -> str:
-        # need to replace `:` with `_` for the project_id; it becomes part of the path and characters
-        # need to work on windows, etc.
-        sanitized_project_id = project.replace(":", "_")
-        return os.path.join(os.path.expanduser(storage_dir), sanitized_project_id)
```

Comment:
issue is that unbeknown to people the project name passed into the tracker gets turned into a path.  so either we enforce what a project name can be, or we still need this and instead save the project name as part of metadata and not rely on the UI to assume that the directory name maps to the project name -- that would be longer term fix.

Reply:
I think it's a bad design assumption on our part to assume the project name is something that becomes part of the file path verbatim. If the project name has `/` in it, etc. then there's more to sanitize, etc.

Reply:
Yeah, its sloppy. As we discussed in person we'll add guards on the project name (fail earlier). 

Not too concerned (the local tracking client is not prod-ready anyway), but we want to reduce onboarding friction.